### PR TITLE
feat(toHaveValue): Asserting aria-valuenow

### LINKED ISSUE #478: Make `.toHaveValue()` check value of `aria-valuenow` as a fallback to `value`
### Describe the feature you'd like:

Suggesting that `.toHaveValue()` support elements' [`aria-valuenow`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-valuenow) attribute

### Suggested implementation:

When element doesn't have the value property, fallback to try and get the `aria-valuenow` attribute

### PR DESCRIPTION
**Pull request by idanen**
Resolves #478

**What**:

Change `.toHaveValue()` to support `aria-valuenow`

**Why**:

**How**:

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Updated Type Definitions
- [x] Ready to be merged

### CODE REVIEW

**Comment by idanen** [comment]
How do I update documentation?

**Review by waynevanson — CHANGES_REQUESTED** [review]

File: src/utils.js [hunk 1]
```diff
@@ -197,18 +197,31 @@ function getInputValue(inputElement) {
...
   switch (element.tagName.toLowerCase()) {
     case 'input':
       return getInputValue(element)
     case 'select':
       return getSelectValue(element)
-    default:
-      return element.value
+    default: {
+      const accessibleValue = getAccessibleValue(element)
+      return element.value ?? accessibleValue
```
[the remark this answers is not in the document]
Comment (idanen) [reply src/utils.js]: `value` should take precedence.
That's how Chrome's accessibility tab behaves

**Review by waynevanson — APPROVED** [review]

**Comment by idanen** [comment]
@gnapse @nickmccurdy any chance one of you can help push this forward?

File: src/utils.js [hunk 2]
```diff
@@ -196,18 +196,30 @@ function getInputValue(inputElement) {
...
 
+const rolesSupportingValues = ['meter', 'progressbar', 'slider', 'spinbutton']
+function getAccessibleValue(element) {
+  if (!rolesSupportingValues.includes(element.getAttribute('role'))) {
+    return
+  }
+  // We want same behavior as accessing the `value` property
+  // eslint-disable-next-line consistent-return
+  return Number(element.getAttribute('aria-valuenow'))
+}
```
Comment (gnapse) [inline src/utils.js:207]: To avoid the eslint issue, maybe you can `return undefined` inside the `if` statement.

**Review by gnapse — APPROVED** [review]
Only one comment that I think needs to be addressed before merging this.

**Comment by idanen** [comment]
@gnapse finally fixed the coverage. Can you please give it another look?

**Comment by gnapse** [comment]
@all-contributors please add @idanen for code, test

**Comment by gnapse** [comment]
@all-contributors please add @waynevanson for code, test

**Comment by gnapse** [comment]
https://github.com/all-contributors please add @waynevanson for code, test

**Comment by gnapse** [comment]
@all-contributors please add @waynevanson for code, test

**Comment by gnapse** [comment]
@all-contributors please add @idanen for code, test

**Pull request merged**